File: EventServiceNodeJS/src/EventMessageInstance.ts

export enum EventMessageType
{
    REQUEST,
    RESPONSE
}

export class EventMessageInstance
{
    public TransactionId: string;
    public Type: EventMessageType;
    public Operation: string;
    public Timestamp: number;
    public Success: boolean;
    constructor(transactionId: string, type: EventMessageType, operation: string, timestamp: number, Success: boolean){
        this.TransactionId = transactionId;
        this.Type = type;
        this.Operation = operation;
        this.Timestamp = timestamp;
        this.Success = Success;
    }
}

File: EventServicePython/src/EventProcessor/EventMessageInstance.py
from enum import Enum, auto


class EventMessageType(Enum):
    REQUEST = 0,
    RESPONSE = 1


class EventMessageInstance:
    def __init__(
        self,
        transaction_id: str,
        message_type: EventMessageType,
        operation: str,
        timestamp: int,
        success: bool

    ):
        self.transaction_id = transaction_id
        self.message_type = message_type
        self.operation = operation
        self.timestamp = timestamp
        self.success = success
